include/data_management/data/soa_numeric_table.h

```diff
@@ -259,14 +250,64 @@ class DAAL_EXPORT SOANumericTable : public NumericTable
     }
 
 protected:
+    /**
+     *  <a name="DAAL-CLASS-DATA_MANAGEMENT__WRAPPEDRAWPOINTER"></a>
+     *  \brief   Class that provides functionality of deep copy.
+     */
+    class WrappedRawPointer
+    {
+    protected:
+        DAAL_INT64 * _arrOffsets;
+        size_t _count;
+
+    public:
+        services::Status allocate(size_t count)
+        {
+            deallocate();
+            _arrOffsets = (DAAL_INT64 *)daal::services::daal_malloc(count * sizeof(DAAL_INT64));
+            DAAL_CHECK_MALLOC(_arrOffsets)
+            _count = count;
+            return services::Status();
+        }
+
+        void deallocate()
+        {
+            if (_arrOffsets)
+            {
+                daal::services::daal_free(_arrOffsets);
+                _arrOffsets = NULL;
+                _count      = 0;
+            }
+        }
+
+        WrappedRawPointer() : _arrOffsets(NULL), _count(0) {};
+        ~WrappedRawPointer() { deallocate(); }
+
+        DAAL_INT64 * get() const { return _arrOffsets; }
```

Comment:
const function returns pointer to non-const memory. Looks like a bug.

Reply:
done